noether_conversions/src/noether_conversions.cpp

```diff
@@ -89,4 +102,195 @@ bool loadPLYFile(const std::string& filename, shape_msgs::Mesh& mesh_msg)
+    a = points[vert[0]].getVector3fMap().cast<double>();
+    b = points[vert[1]].getVector3fMap().cast<double>();
+    c = points[vert[2]].getVector3fMap().cast<double>();
+    dir = (b - a).cross((c -a)).normalized();
+    dir = flip ? -1.0 * dir : dir;
+
+    // assigning to points
+    for(std::size_t v = 0; v < vert.size(); v++)
+    {
+      PointNormal& p = cloud_normals[vert[v]];
+      p.normal_x = dir.x();
+      p.normal_y = dir.y();
+      p.normal_z = dir.z();
+    }
+  }
+}
+
+visualization_msgs::MarkerArray convertToAxisMarkers(const noether_msgs::ToolRasterPath& toolpath,
+                                                     const std::string& frame_id, const std::string& ns,
+                                                     const std::size_t start_id, const double axis_scale,
+                                                     const double axis_length,
+                                                     const std::tuple<float, float, float, float, float, float>& offset)
+{
+  return convertToAxisMarkers(toolpath.paths, frame_id, ns, start_id, axis_scale, axis_length, offset);
+}
+
+visualization_msgs::MarkerArray convertToAxisMarkers(const std::vector<geometry_msgs::PoseArray>& path,
+                                                     const std::string& frame_id, const std::string& ns,
+                                                     const std::size_t start_id, const double axis_scale,
+                                                     const double axis_length,
+                                                     const std::tuple<float, float, float, float, float, float>& offset)
+{
+  using namespace Eigen;
+
+  visualization_msgs::MarkerArray markers;
+
+  auto create_line_marker = [&](const int id, const std::tuple<float, float, float, float>& rgba) -> visualization_msgs::Marker
+  {
+    visualization_msgs::Marker line_marker;
+    line_marker.action = line_marker.ADD;
+    std::tie(line_marker.color.r, line_marker.color.g, line_marker.color.b, line_marker.color.a) = rgba;
+    line_marker.header.frame_id = frame_id;
+    line_marker.type = line_marker.LINE_LIST;
+    line_marker.id = id;
+    line_marker.lifetime = ros::Duration(0);
+    line_marker.ns = ns;
+    std::tie(line_marker.scale.x, line_marker.scale.y, line_marker.scale.z) = std::make_tuple(axis_scale, 0.0, 0.0);
+    line_marker.pose = pose3DtoPoseMsg(offset);
+    return std::move(line_marker);
+  };
+
+  // markers for each axis line
+  int marker_id = start_id;
+  visualization_msgs::Marker x_axis_marker = create_line_marker(++marker_id,std::make_tuple(1.0, 0.0, 0.0, 1.0));
+  visualization_msgs::Marker y_axis_marker = create_line_marker(++marker_id,std::make_tuple(0.0, 1.0, 0.0, 1.0));
+  visualization_msgs::Marker z_axis_marker = create_line_marker(++marker_id,std::make_tuple(0.0, 0.0, 1.0, 1.0));
+
+  auto add_axis_line = [](const Isometry3d& eigen_pose, const Vector3d& dir,
```

Comment:
I recommend making this a separate function with documentation. Having a lambda function this big within another function affects the readability of the function you are within.

Reply:
It really isn't that big a lambda function, plus I wanted to avoid creating a global function that isn't reusable

Reply:
The point I was making is that readability is important and this makes it difficult to read for outsiders looking in. Also per [C++ Core Guidlines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#Rf-capture-vs-overload) states that in this use case it should be enforced that an ordinary function be used.